rules/awsrules/aws_resource_tags.go

```diff
@@ -0,0 +1,311 @@
+		"aws_redshift_snapshot_schedule",
+		"aws_redshift_subnet_group",
+		"aws_resourcegroups_group",
+		"aws_route53_health_check",
+		"aws_route53_resolver_endpoint",
+		"aws_route53_resolver_rule",
+		"aws_route53_zone",
+		"aws_route_table",
+		"aws_s3_bucket",
+		"aws_s3_bucket_object",
+		"aws_sagemaker_endpoint",
+		"aws_sagemaker_endpoint_configuration",
+		"aws_sagemaker_model",
+		"aws_sagemaker_notebook_instance",
+		"aws_secretsmanager_secret",
+		"aws_security_group",
+		"aws_servicecatalog_portfolio",
+		"aws_sfn_activity",
+		"aws_sfn_state_machine",
+		"aws_sns_topic",
+		"aws_spot_instance_request",
+		"aws_sqs_queue",
+		"aws_ssm_activation",
+		"aws_ssm_document",
+		"aws_ssm_maintenance_window",
+		"aws_ssm_parameter",
+		"aws_ssm_patch_baseline",
+		"aws_storagegateway_cached_iscsi_volume",
+		"aws_storagegateway_gateway",
+		"aws_storagegateway_nfs_file_share",
+		"aws_storagegateway_smb_file_share",
+		"aws_subnet",
+		"aws_swf_domain",
+		"aws_transfer_server",
+		"aws_transfer_user",
+		"aws_vpc",
+		"aws_vpc_dhcp_options",
+		"aws_vpc_endpoint",
+		"aws_vpc_endpoint_service",
+		"aws_vpc_peering_connection",
+		"aws_vpc_peering_connection_accepter",
+		"aws_vpn_connection",
+		"aws_vpn_gateway",
+		"aws_waf_rate_based_rule",
+		"aws_waf_rule",
+		"aws_waf_rule_group",
+		"aws_waf_web_acl",
+		"aws_wafregional_rate_based_rule",
+		"aws_wafregional_rule",
+		"aws_wafregional_rule_group",
+		"aws_wafregional_web_acl",
+		"aws_workspaces_directory",
+		"aws_workspaces_ip_group",
+	}
+	return &AwsResourceTagsRule{
+		resourceTypes: resourceTypes,
+	}
+}
+
+// Name returns the rule name
+func (r *AwsResourceTagsRule) Name() string {
+	return "aws_resource_tags"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsResourceTagsRule) Enabled() bool {
+	return true
+}
+
+// Severity returns the rule severity
+func (r *AwsResourceTagsRule) Severity() string {
+	return tflint.ERROR
+}
+
+// Link returns the rule reference link
+func (r *AwsResourceTagsRule) Link() string {
+	return tflint.ReferenceLink(r.Name())
+}
+
+// Check checks for matching tags
+func (r *AwsResourceTagsRule) Check(runner *tflint.Runner) error {
+	config :=  awsResourceTagsRuleConfig{}
+	if err := runner.DecodeRuleConfig(r.Name(), &config); err != nil {
+		return err
+	}
+
+	for _, resourceType := range r.resourceTypes {
+		err := runner.WalkResourceAttributes(resourceType, "tags", func(attribute *hcl.Attribute) error {
+			var resourceTags map[string]string
+			err := runner.EvaluateExpr(attribute.Expr, &resourceTags)
+			tags := []string{}
+			for k := range resourceTags {
+				tags = append(tags, k)
+			}
+
+			return runner.EnsureNoError(err, func() error {
+				hash := make(map[string]bool)
+				for _, k := range tags {
+					hash[k] = true
+				}
+				var found []string
+				for _, tag := range config.Tags {
+					if _, ok := hash[tag]; ok {
+						found = append(found, tag)
+					}
+				}
+				if len(found) != len(config.Tags) {
```

Comment:
In this case, the rule will emit an issue if the resource has extra tags. Intuitively, if it has been tagged with configured tags, I think that other tags should be allowed, but what about?

Reply:
Actually, the logic I had there was correct, if a bit confusing. The `found` slice only contained the subset of tags that had been found in the config. If the length of the slices didn't match, it meant that some of the expected tags were missing. 

However, I've updated the logic in the most recent commit and it's now checked differently anyway, hopefully in a way that is more explicit.